src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,146 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        Node(Node<T> prev, T item, Node<T> next) {
+            this.prev = prev;
+            this.item = item;
+            this.next = next;
+        }
+    }
+
+    private T unlink(Node<T> deletedNode) { // Для зміни зв'язків
+        Node<T> nextNode = deletedNode.next;
+        Node<T> prevNode = deletedNode.prev;
+        if (prevNode == null) {
+            first = nextNode;
+        } else {
+            prevNode.next = nextNode;
+        }
+
+        if (nextNode == null) {
+            last = prevNode;
+        } else {
+            nextNode.prev = prevNode;
+        }
+        size--;
+        return deletedNode.item;
+    }
+
+    private void checkPosition(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index is invalid!");
+        }
+    }
+
+    private Node<T> findNode(int index) {
+        Node<T> findNode;
+        if (index < size / 2) {
+            findNode = first;
+            for (int i = 1; i <= index; i++) {
+                findNode = findNode.next;
+            }
+        } else {
+            findNode = last;
+            for (int i = size - 2; i >= index; i--) {
+                findNode = findNode.prev;
+            }
+        }
+        return findNode;
+    }
+
     @Override
     public void add(T value) {
+        final Node<T> lastNode = last;
+        final Node<T> newNode = new Node<>(lastNode, value, null);
+        last = newNode;
+        if (lastNode == null) {
+            first = newNode;
+        } else {
+            lastNode.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        //checkPosition(index - 1);
```

Comment:
if we have 9 elements in list, why can't we insert on index 9?

Reply:
In checkPosition() method, index check '>='.
I did the same checking, in add() it was working, but in checkPosition is wasn't working.